File: Test/Huella_CO2.go
package main

import (
	"fmt"
)

const (
	FACTOR_CARRO     = 121.0
	FACTOR_MOTO      = 53.0
	FACTOR_BICICLETA = 3.0

	FACTOR_BUS = 49.0

	FACTOR_CARNE_ROJA  = 7.19
	FACTOR_PESCADO     = 3.91
	FACTOR_VEGETARIANA = 3.81
)

func main() {
	mostrarMensaje("KILÓMETROS RECORRIDOS EN VEHÍCULOS PARTICULARES")
	kmCarro := ingresarReal("- carro: ")
	kmMoto := ingresarReal("- moto: ")
	kmBicicleta := ingresarReal("- bicicleta: ")

	mostrarMensaje("\nKILÓMETROS RECORRIDOS EN TRANSPORTE URBANO")
	trayectosBus := ingresarReal("- bus: ")

	mostrarMensaje("\nCANTIDAD COMIDAS CON")
	comidasCarneRoja := ingresarEntero("- carne roja: ")
	comidasPescado := ingresarEntero("- pescado: ")
	comidasVegetarianas := ingresarEntero("- vegetarianas: ")

	kgCO2Recorridos := calcularKgCO2Recorridos(
		kmCarro,
		kmMoto,
		kmBicicleta)

	kgCO2Trayectos := calcularKgCO2Trayectos(trayectosBus)

	kgCO2Comidas := calcularKgCO2Comidas(
		comidasCarneRoja,
		comidasPescado,
		comidasVegetarianas)

	kgCO2Total := calcularKgCO2Total(
		kgCO2Recorridos,
		kgCO2Trayectos,
		kgCO2Comidas)

	mensajeSalida := generarMensajeSalida(
		kgCO2Recorridos,
		kgCO2Trayectos,
		kgCO2Comidas,
		kgCO2Total)

	mostrarMensaje(mensajeSalida)
}

func mostrarMensaje(mensajeSalida string) {
	fmt.Println(mensajeSalida)
}

func ingresarReal(pregunta string) float64 {
	var real float64
	fmt.Print(pregunta)
	fmt.Scanln(&real)
	return real
}

func ingresarEntero(pregunta string) int {
	var entero int
	fmt.Print(pregunta)
	fmt.Scanln(&entero)
	return entero
}

func calcularKgCO2Recorridos(
	kmCarro,
	kmMoto,
	kmBicicleta float64) float64 {
	return kmCarro*FACTOR_CARRO + kmMoto*FACTOR_MOTO +
		kmBicicleta*FACTOR_BICICLETA
}

func calcularKgCO2Trayectos(trayectosBus float64) float64 {
	return float64(trayectosBus) * FACTOR_BUS
}

func calcularKgCO2Comidas(
	comidasCarneRoja,
	comidasCarneBlanca,
	comidasVegetarianas int) float64 {
	return float64(comidasCarneRoja)*FACTOR_CARNE_ROJA +
		float64(comidasCarneBlanca)*FACTOR_PESCADO +
		float64(comidasVegetarianas)*FACTOR_VEGETARIANA
}

func calcularKgCO2Total(
	kgCO2Recorridos,
	kgCO2Trayectos,
	kgCO2Comidas float64) float64 {
	return kgCO2Recorridos + kgCO2Trayectos + kgCO2Comidas
}

func generarMensajeSalida(kgCO2Recorridos,
	kgCO2Trayectos,
	kgCO2Comidas,
	kgCO2Total float64) string {
	return fmt.Sprintf("\nSu huella de carbono en kg de CO2 por:\n"+
		"- RECORRIDOS %.2f kg\n"+
		"- TRAYECTOS %.2f kg\n"+
		"- COMIDAS %.2f kg\n"+
		"\nPara un TOTAL de %.2f kg de emisiones de CO2.",
		kgCO2Recorridos, kgCO2Trayectos, kgCO2Comidas, kgCO2Total)
}

File: Test/Huella_CO2.py
FACTOR_CARRO = 121.0
FACTOR_MOTO = 53.0
FACTOR_BICICLETA = 3.0

FACTOR_BUS = 49.0

FACTOR_CARNE_ROJA = 7.19
FACTOR_PESCADO = 3.91
FACTOR_VEGETARIANA = 3.81


def main():
    mostrar_mensaje("KILÓMETROS RECORRIDOS EN VEHÍCULOS PARTICULARES")
    km_carro = ingresar_real("- carro: ")
    km_moto = ingresar_real("- moto: ")
    km_bicicleta = ingresar_real("- bicicleta eléctrica: ")

    mostrar_mensaje("\nKILÓMETROS RECORRIDOS EN TRANSPORTE URBANO")
    trayectos_bus = ingresar_real("- bus: ")

    mostrar_mensaje("\nCANTIDAD COMIDAS CON")
    comidas_carne_roja = ingresar_entero("- carne roja: ")
    comidas_pescado = ingresar_entero("- pescado: ")
    comidas_vegetarianas = ingresar_entero("- vegetarianas: ")

    kg_co2_recorridos = calcular_kg_co2_recorridos(km_carro,
                                                   km_moto,
                                                   km_bicicleta)

    kg_co2_trayectos = calcular_kg_co2_trayectos(trayectos_bus)

    kg_co2_comidas = calcular_kg_co2_comidas(comidas_carne_roja,
                                             comidas_pescado,
                                             comidas_vegetarianas)

    kg_co2_total = calcular_kg_co2_total(kg_co2_recorridos,
                                         kg_co2_trayectos,
                                         kg_co2_comidas)

    mensaje_salida = generar_mensaje_salida(kg_co2_recorridos,
                                            kg_co2_trayectos,
                                            kg_co2_comidas,
                                            kg_co2_total)

    mostrar_mensaje(mensaje_salida)


def mostrar_mensaje(mensaje):
    print(mensaje)


def ingresar_texto(pregunta):
    return input(pregunta)


def ingresar_entero(pregunta):
    return int(ingresar_texto(pregunta))


def ingresar_real(pregunta):
    return float(ingresar_texto(pregunta))


def calcular_kg_co2_recorridos(km_carro,
                               km_moto,
                               km_bicicleta):
    return (
        km_carro * FACTOR_CARRO +
        km_moto * FACTOR_MOTO +
        km_bicicleta * FACTOR_BICICLETA
    )


def calcular_kg_co2_trayectos(trayectos_bus):
    return trayectos_bus * FACTOR_BUS


def calcular_kg_co2_comidas(comidas_carne_roja,
                            comidas_carne_blanca,
                            comidas_vegetarianas):
    return (
        comidas_carne_roja * FACTOR_CARNE_ROJA +
        comidas_carne_blanca * FACTOR_PESCADO +
        comidas_vegetarianas * FACTOR_VEGETARIANA
    )


def calcular_kg_co2_total(kg_co2_recorridos,
                          kg_co2_trayectos,
                          kg_co2_comidas):
    return kg_co2_recorridos + kg_co2_trayectos + kg_co2_comidas


def generar_mensaje_salida(kg_co2_recorridos,
                           kg_co2_trayectos,
                           kg_co2_comidas,
                           kg_co2_total):
    return (
        f"\nSu huella de carbono en kg de CO2 por:"
        f"\n- RECORRIDOS {kg_co2_recorridos:.2f} kg"
        f"\n- TRAYECTOS {kg_co2_trayectos:.2f} kg"
        f"\n- COMIDAS {kg_co2_comidas:.2f} kg"
        f"\n\nPara un TOTAL de {kg_co2_total:.2f} "
        f"kg de emisiones de CO2."
    )


main()
